pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -529,16 +531,30 @@ func getPodCIDRsOnNode(node *corev1.Node) []string {
 // createIPSecTunnelPort creates an IPSec tunnel port for the remote Node if the
 // tunnel does not exist, and returns the ofport number.
 func (c *Controller) createIPSecTunnelPort(nodeName string, nodeIP net.IP) (int32, error) {
+	portName := util.GenerateNodeTunnelInterfaceName(nodeName)
 	interfaceConfig, ok := c.interfaceStore.GetNodeTunnelInterface(nodeName)
-	if ok {
-		// TODO: check if Node IP, PSK, or tunnel type changes. This can
-		// happen if removeStaleTunnelPorts fails to remove a "stale"
-		// tunnel port for which the configuration has changed.
+	stalePortRemoved := false
+	// check if Node IP, PSK, or tunnel type changes. This can
+	// happen if removeStaleTunnelPorts fails to remove a "stale"
+	// tunnel port for which the configuration has changed.
+	if ok && (interfaceConfig.InterfaceName != portName ||
+		interfaceConfig.PSK != c.networkConfig.IPSecPSK ||
+		!interfaceConfig.RemoteIP.Equal(nodeIP) ||
+		interfaceConfig.TunnelInterfaceConfig.Type != c.networkConfig.TunnelType) {
```

Comment:
this seems to be duplicated from `removeStaleTunnelPorts`, can we define a separate function for tunnel config comparison?

Reply:
sure, will do.